StitchVolumes/StitchVolumes.py

```diff
@@ -0,0 +1,706 @@
+        # self._parameterNode.SetNodeReferenceID("OutputVolume", self.ui.outputSelector.currentNodeID)
+        # self._parameterNode.SetParameter("Threshold", str(self.ui.imageThresholdSliderWidget.value))
+        # self._parameterNode.SetParameter("Invert", "true" if self.ui.invertOutputCheckBox.checked else "false")
+        # self._parameterNode.SetNodeReferenceID("OutputVolumeInverse", self.invertedOutputSelector.currentNodeID)
+
+    def onApplyButton(self):
+        """
+        Run processing when user clicks "Stitch Volumes" button.
+        """
+        try:
+            # Gather inputs
+            orig_nodes = self.gather_original_nodes()
+            roi_node = self.ui.roiSelector.currentNode()
+            stitched_vol_name = self.ui.stitchVolNameLineEdit.text
+            # Run the stitching
+            self.logic.stitch_volumes(
+                orig_nodes, roi_node, stitched_vol_name, keep_intermediate_volumes=False
+            )
+
+        except Exception as e:
+            slicer.util.errorDisplay("Failed to compute results: " + str(e))
+            import traceback
+
+            traceback.print_exc()
+
+    def gather_original_nodes(self):
+        orig_nodes = []
+        if self.ui.volumeSelector1.currentNode():
+            orig_nodes.append(self.ui.volumeSelector1.currentNode())
+        if self.ui.volumeSelector2.currentNode():
+            orig_nodes.append(self.ui.volumeSelector2.currentNode())
+        if self.ui.volumeSelector3.currentNode():
+            orig_nodes.append(self.ui.volumeSelector3.currentNode())
+        if self.ui.volumeSelector4.currentNode():
+            orig_nodes.append(self.ui.volumeSelector4.currentNode())
+        if self.ui.volumeSelector5.currentNode():
+            orig_nodes.append(self.ui.volumeSelector5.currentNode())
+        return orig_nodes
+
+
+#
+# StitchVolumesLogic
+#
+
+
+class StitchVolumesLogic(ScriptedLoadableModuleLogic):
+    """This class should implement all the actual
+    computation done by your module.  The interface
+    should be such that other python code can import
+    this class and make use of the functionality without
+    requiring an instance of the Widget.
+    Uses ScriptedLoadableModuleLogic base class, available at:
+    https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def setDefaultParameters(self, parameterNode):
+        """
+        Initialize parameter node with default settings.
+        """
+        if not parameterNode.GetParameter("OutputVolName"):
+            parameterNode.SetParameter("OutputVolName", "S")
+
+    def stitch_volumes(
+        self, orig_nodes, roi_node, stitched_vol_name, keep_intermediate_volumes=False
+    ):
+        # Stitch together the supplied original volumes, resampling them
+        # into the space defined by the supplied roi, putting the stitched
+        # output into a volume with the given stitched volume name
```

Comment:
Now that you add this nice and convenient module, it would be great if you could also make the algorithm a bit more sophisticated by computing weighted average of the volumes, using the distance from the volume edge (can be computed quickly using vtkITKDistanceTransform) as weighting factor.

Reply:
It makes sense to add this as an option.  I would definitely retain the option to have a hard border to be able to easily evaluate alignment at the seams.  In addition, I would need to think about whether this could lead to averaging of image data with fill value data or not.